packages/protocol/contracts/common/UsingPrecompiles.sol

```diff
@@ -140,4 +142,60 @@ contract UsingPrecompiles {
     );
     return success;
   }
+
+  /**
+   * @notice Parses block number out of header.
+   * @param header RLP encoded header
+   * @return Block number.
+   */
+  function getBlockNumberFromHeader(bytes memory header) public returns (uint256) {
+    bytes memory blockNumber;
+    bool success;
+    (success, blockNumber) = GET_BLOCK_NUMBER_FROM_HEADER.call.value(0).gas(gasleft())(
```

Comment:
I believe we should use `staticcall` here. See https://solidity.readthedocs.io/en/v0.5.0/types.html#address